src/api/posts.rs

```diff
@@ -0,0 +1,21 @@
+use canapi::Provider;
+use diesel::PgConnection;
+use rocket_contrib::Json;
+use serde_json;
+
+use plume_api::posts::PostEndpoint;
+use plume_models::db_conn::DbConn;
+use plume_models::posts::Post;
+
+#[get("/posts/<id>")]
+fn get(id: i32, conn: DbConn) -> Json<serde_json::Value> {
+    let post = <Post as Provider<PgConnection>>::get(&*conn, id).ok();
+    Json(json!(post))
+}
+
+// TODO: handle query params
```

Comment:
such as?

Reply:
For instance, if you `GET /api/v1/posts?title=Hello%20world`, only posts with the `Hello world` title should be shown (`title` is not the best example, but `author_id` could be interesting for instance)

Reply:
So is this (at least partially) blocked by #149, or not related?

Reply:
No, actually we just need to get the raw query params, and deserialize them to `PostEndpoint` with something like `serde_qs`.